docs/source/newsfragments/4622.change.rst

```diff
@@ -0,0 +1 @@
+:envvar:`COCOTB_RESOLVE_X` behavior was changed so that :class:`.LogicArray` and :class:`.Logic` values are resolved as soon as they are returned from the :meth:`handle.get() <cocotb.handle.ValueObjectBase.get>` or :attr:`handle.value <cocotb.handle.ValueObjectBase.value>`, and not only when the value is converted to an :class:`int`. It also only affects values return from handles and not all :class:`!.LogicArray`\ s or :class:`!.Logic`\ s.
```

Comment:
```suggestion
:envvar:`COCOTB_RESOLVE_X` behavior was changed so that :class:`.LogicArray` and :class:`.Logic` values are resolved as soon as they are returned from the :meth:`handle.get() <cocotb.handle.ValueObjectBase.get>` or :attr:`handle.value <cocotb.handle.ValueObjectBase.value>`, and not only when the value is converted to an :class:`int`. It also only affects values returned from handles and not all :class:`!.LogicArray`\ s or :class:`!.Logic`\ s.
```